### ISSUE TITLE
October release

### ISSUE THREAD

**Issue by damienmg** [bazelbuild/bazel#1902]
0.3.2 is finally out so creating that bug to track progress next release.

We should probably wait a bit before cutting a new release. Ideally October release will be 0.4.

Assigning to Kristina, this month release manager :)

**Comment by kchodorow** [comment]
So, CI still isn't green.  Seems like I should wait until it is, so... on hold.

**Comment by damienmg** [comment]
We are running down blocker for 0.4 so it might be useful to wait until these get resolved.

**Comment by damienmg** [comment]
FWIW except the regression I reported today as release blocker, all non documentation issue have inflight change so we can definitely announce this one as 0.4 :)

**Comment by kchodorow** [comment]
Weekly update: CI still isn't green.

**Comment by kchodorow** [comment]
And off we go:

```
$ scripts/release/release.sh create 0.4.0 088bbc614c184822050dfa5768d8c1281458c275 b01160c4bbfa350640a84ac8d55d79d035c72308
$ scripts/release/release.sh push
```

The cherry pick is https://bazel-review.googlesource.com/c/6832/ (stamping for the Windows release).

CI run: http://ci.bazel.io/view/Bazel%20bootstrap%20and%20maintenance/job/Bazel/932/.

**Comment by kchodorow** [comment]
Failed on ci because gpg was trying to [talk to a tty](http://ci.bazel.io/view/Bazel%20bootstrap%20and%20maintenance/job/Bazel-Release/78/console).  Cherry picked 2d6736e519a143f4d9c64bb55001401fcf380120 to fix.

**Comment by kchodorow** [comment]
Still having GPG issues: http://ci.bazel.io/view/Bazel%20bootstrap%20and%20maintenance/job/Bazel-Release/79/console

**Comment by meteorcloudy** [comment]
Please cherry pick https://github.com/bazelbuild/bazel/commit/81aede1f9bdddfe0e2327d6cf0ba9bd40fd62718 :)
It's a fix for building Tensorflow

**Comment by kchodorow** [comment]
Pushed:

```
$ scripts/release/release.sh create 0.4.0 088bbc614c184822050dfa5768d8c1281458c275 b01160c4bbfa350640a84ac8d55d79d035c72308 2d6736e519a143f4d9c64bb55001401fcf380120 9b1dfb8 81aede1
```

Includes: 

* whole archive fix.
* second gpg fix.

You know what super annoying? Having to re-edit the release notes every time I cherry pick something.  Let me see if I can fix that (in my copious free time).

**Comment by meteorcloudy** [comment]
Thanks! We finally have a release candidate now! \o/

**Comment by petemounce** [comment]
@meteorcloudy would you like a chocolatey pre-release package for it? Requirement for that would be to push the RC (specifically, the zip that the chocolatey package depends on) to github releases (or, some other place - I can code around the URI being a different string).

**Comment by meteorcloudy** [comment]
@petemounce That would be awesome! Every thing we build for the release candidate is here [http://storage.googleapis.com/bazel](http://storage.googleapis.com/bazel)
I have manually upload a zip file for 0.4.0rc3, full url is [http://storage.googleapis.com/bazel/0.4.0/rc3/bazel-0.4.0rc3-windows-x86_64.zip](http://storage.googleapis.com/bazel/0.4.0/rc3/bazel-0.4.0rc3-windows-x86_64.zip)
I also updated our release process to automatically release a zip file. :)

**Comment by petemounce** [comment]
@meteorcloudy I'll do that shortly. PR incoming.

It would be convenient if the storage bucket also contained the sha256 file for the zip so I can use that similar to how I create the release package ([ref](https://github.com/bazelbuild/bazel/blob/master/scripts/packages/chocolatey/build.ps1#L23)).

Also, the automation at your end that generates the index.html does not yet include a link to the zip file + sha256 (I initially went there and thought there was no zip until I tried you direct link).

**Comment by petemounce** [comment]
It turns out powershell's get-filehash cmdlet is very slow compared to msys2's - the former is still going after 15 minutes and 1/3 progress, compared to msys2's finishing in less time than it took to type this. Getting the signature file to be published with the rc would be very welcome, since otherwise the build.ps1 will need to take a dependency on msys2, which seems overkill for one command.

**Comment by petemounce** [comment]
The RC package is published.

**Comment by laszlocsomor** [comment]
The release should include the fix for https://github.com/bazelbuild/bazel/issues/1463. Can we cut from a later commit than the current baseline, e.g. https://github.com/bazelbuild/bazel/commit/ca99bb71b8120e61a3bbde8e54f1a9488fa0478f?

**Comment by kchodorow** [comment]
Whoops, didn't see @laszlocsomor's comment.  Release is now available at https://github.com/bazelbuild/bazel/releases/tag/0.4.0.  Guess we should get 0.4.1 right out.

**Comment by damienmg** [comment]
I don't think a P1 should mandate a new release.

**Comment by petemounce** [comment]
@meteorcloudy @damienmg @kchodorow As soon as you publish the zip (and sha256 of it) that the chocolatey package requires to github, I can publish the 0.4.0 chocolatey package that depends on it. I suppose I could go from the google storage bucket like the RC, but I'd prefer to continue to use the github releases source.

**Comment by petemounce** [comment]
@kchodorow re: 0.4.1 - I'd love to have a version that has the windows-c-drive fix asap, but I don't know what overhead doing a release has to know whether that's reasonable from your end.

**Comment by damienmg** [comment]
We have to do a release this month so I think starting a release candidate
next week is reasonable. Nothing prevent us from going faster on release
though.

**Comment by meteorcloudy** [comment]
@petemounce The zip file of 0.4.0 and it's sha256 are available on github now! :) 
For the later release, everything for release candidate will automatically be available in google storage bucket, and everything for real release will be available in github releases source.

Thanks again!

**Comment by petemounce** [comment]
@meteorcloudy thanks! https://chocolatey.org/packages/bazel/0.4.0 has been published (`choco install -v 0.4.0` ought to work), and it's in their moderation queue.
